src/db/utils/get-thread-by-id/get-thread-by-id.ts

```diff
@@ -0,0 +1,9 @@
+import { ThreadModel } from "../../../models/thread/thread.model";
+/**
+ * Finds user by ID and only returns relevant properties
+ * @param userId ObjectId
+ */
+export async function getThreadById({threadId}: {threadId: string}) {
+  const result = await ThreadModel.findById(threadId);
+  return result;
```

Comment:
I'm wondering. Can we just not use the `ThreadModel.findById` static method directly? Maybe there is a reason why we need a wrapper method?

Reply:
So as it is, yes that makes sense, though I've a draft 114 (114-refactor though not strictly just a refactor now) I'd still like to merge in at some point, which for this does a little more so the client can do a little less... 
`export async function getThreadById({threadId, reqUserId}: {threadId: string, reqUserId: string}): Promise<IThreadResponse> {
  const result = await ThreadModel.findById(threadId);
  if (!result) throw new Error(`Unable to find thread for id: ${threadId}`);
  const currentUserReactions: any = {};
  const reactionsAsArray = Object.entries(result.reactions);
  reactionsAsArray.forEach(([key, value]) => {
    if (value.postedByUserId.toString() === reqUserId) {
      currentUserReactions[key] = value;
    }
  })
  return {
    id: result._id.toString(),
    postedByUserId: result.postedByUserId.toString(),
    visibility: result.visibility,
    content: result.content,
    comments: Object.values(result.comments).sort((a, b) => b.updatedAt.valueOf() - a.updatedAt.valueOf()),
    reactions: result.reactions,
    reactionsCount: reactionsAsArray.length,
    currentUserReactions,
    forks: result.forks,
    aForkOfThreadId: result.aForkOfThreadId,
    updatedAt: result.updatedAt.valueOf().toString(),
    createdAt: result.createdAt.valueOf().toString(),
  };
}`

Reply:
ahh ok! Makes sense!